pkg/autoscaler/autoscaler/external_autoscaler.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package autoscaler
+
+import (
+	"fmt"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/klog"
+)
+
+const (
+	// The TidbCluster for the external query will be "<original-tcname>-<component>-external"
+	externalTcNamePattern = "%s-%s-external"
+	specialUseLabelKey    = "specialUse"
+	specialUseHotRegion   = "hotRegion"
+)
+
+func (am *autoScalerManager) syncExternalResult(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler, component v1alpha1.MemberType, targetReplicas int32) error {
+	externalTcName := fmt.Sprintf(externalTcNamePattern, tc.ClusterName, component.String())
+	externalTc, err := am.tcLister.TidbClusters(tc.Namespace).Get(externalTcName)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			if targetReplicas <= 0 {
+				return nil
+			}
+			return am.createExternalAutoCluster(tc, externalTcName, tac, component, targetReplicas)
+		}
+
+		klog.Errorf("tac[%s/%s] failed to get external tc[%s/%s], err: %v", tac.Namespace, tac.Name, tc.Namespace, externalTcName, err)
+		return err
+	}
+
+	if targetReplicas <= 0 {
+		err := am.cli.PingcapV1alpha1().TidbClusters(externalTc.Namespace).Delete(externalTc.Name, nil)
```

Comment:
This will cause all the TiKV Pods to be deleted at the same time without transferring data, we should scale the replicas to 0 first and then wait until all of the Pods created for this TidbCluster have been deleted, then we can delete the TidbClusters.

Reply:
The same logic should apply to the deletion of TCs created with PD API.